redis/ocsp.py

```diff
@@ -11,8 +11,50 @@
 from redis.exceptions import AuthorizationError, ConnectionError
 
 
+def _check_certificate(certificate):
+    """A wrapper the return the validity of a known ocsp certificate"""
+
+    ocsp_response = ocsp.load_der_ocsp_response(certificate)
+    if ocsp_response.response_status == ocsp.OCSPResponseStatus.UNAUTHORIZED:
+        raise AuthorizationError("you are not authorized to view this ocsp certificate")
+    if ocsp_response.response_status == ocsp.OCSPResponseStatus.SUCCESSFUL:
+        if ocsp_response.certificate_status == ocsp.OCSPCertStatus.REVOKED:
+            return False
+        else:
+            return True
+    else:
+        return False
+
+
+def ocsp_staple_verifier(con, ocsp_bytes, expected):
+    """An implemention of a function for set_ocsp_client_callback in PyOpenSSL.
+
+    This function validates that the provide ocsp_bytes response is valid,
+    and matches the expected, stapled responses.
+    """
+    if ocsp_bytes in [b"", None]:
+        raise ConnectionError("no ocsp response present")
+
+    issuer_cert = None
+    peer_cert = con.get_peer_certificate().to_cryptography()
+    for c in con.get_peer_cert_chain():
+        if c.to_cryptography().subject == peer_cert.issuer:
+            issuer_cert = c
+            break
+
+    if issuer_cert is None:
+        raise ConnectionError("no matching issuer cert found in certificate chain")
+
+    e = x509.load_pem_x509_certificate(expected)
```

Comment:
This will fail if user does not provide an expected certificate. do we want to allow users not to provide one? I think so, that's up to a user to decide if he needs it